utils/archive_build_size.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python
+"""Track changes in the size of an image using Travis-CI and GitHub statuses.
+
+This code is adapted from https://github.com/danvk/travis-weigh-in
+
+Usage:
+    archive_build_size.py <MAKETARGET>
+
+This should be run inside of a Travis-CI worker.
+It will post the current size of the file as a GitHub status on the commit.
+
+If it's able to deduce the size of the file before the change, it will report
+the size delta. This requires that this script was run on the base for a Pull
+Request, e.g. the commit that was merged into master.
+"""
+
+import os
+import re
+import sys
+import json
+import urllib2
+from collections import namedtuple
+
+TRAVIS = os.environ.get('TRAVIS')
+TRAVIS_COMMIT = os.environ.get('TRAVIS_COMMIT')
+TRAVIS_PULL_REQUEST = os.environ.get('TRAVIS_PULL_REQUEST')
+TRAVIS_REPO_SLUG = os.environ.get('TRAVIS_REPO_SLUG')
+GITHUB_TOKEN = os.environ.get('GITHUB_TOKEN')
+
+ImageSize = namedtuple('ImageSize', ['rom', 'ram'])
+
+if TRAVIS_PULL_REQUEST == 'false':
+    TRAVIS_PULL_REQUEST = False
+
+# The PR job has all the needed info to compute deltas.
+# But GitHub shows statuses for the commit associated with the push job.
+# For the PR job, this will be the status URL for the push job.
+TRAVIS_STATUS_URL = None
+
+
+def raise_for_status(url, response):
+    """Raise exception on URL error"""
+    if response.getcode() < 200 or response.getcode() >= 300:
+        sys.stderr.write(response.read())
+        sys.stderr.write('\n')
+        raise Exception('Request for %s failed: %s' % (url, response.getcode()))
+
+
+def post_status(url, state, context, description):
+    """Send status update to GitHub"""
+    data = {
+        'state': state,
+        'context': context,
+        'description': description
+    }
+    headers = {'Authorization': 'token ' + GITHUB_TOKEN}
+
+    request = urllib2.Request(url, json.dumps(data), headers)
+    res = urllib2.urlopen(request)
+    raise_for_status(url, res)
+
+    print 'Posted %s' % json.dumps(data)
+
+
+def get_status(url, context):
+    """Request status for previous build from GitHub"""
+    headers = {'Authorization': 'token ' + GITHUB_TOKEN}
+    request = urllib2.Request(url, None, headers)
+    res = urllib2.urlopen(request)
+    raise_for_status(url, res)
+
+    data = json.loads(res.read())
+    for status in data:
+        if status['context'] == context:
+            return status['description']
+    return None
+
+
+def get_pr_info(slug, pull_number):
+    """Retireve pull-request info"""
+    url = 'https://api.github.com/repos/%s/pulls/%s' % (slug, pull_number)
+    headers = {'Authorization': 'token ' + GITHUB_TOKEN}
+    request = urllib2.Request(url, None, headers)
+    res = urllib2.urlopen(request)
+    raise_for_status(url, res)
+    return json.loads(res.read())
+
+
+def get_context(filename):
+    """Define status context"""
+    return "image_size/" + filename
```

Comment:
Can we shorten image_size so the status does not get truncated in pull requests overview?

Reply:
Agree. We can display delta first then the size. So we can at least see the diff without mouse hover. We can remove unit b as well.

Reply:
I removed the 'image_size/' part and the 'b'.  That seems to provide enough space